pkg/api/endpoints/util.go

```diff
@@ -83,21 +65,63 @@ func RepackSubsets(subsets []api.EndpointSubset) []api.EndpointSubset {
 	// Next, build the N-to-M association the API wants.
 	final := []api.EndpointSubset{}
 	for key, ports := range addrSetsToPorts {
-		addrs := []api.EndpointAddress{}
-		for k := range addrSets[key] {
-			addrs = append(addrs, *k)
+		var addrs, notReadyAddrs []api.EndpointAddress
+		for k, ready := range addrSets[key] {
```

Comment:
I feel like I'm taking Tim's code review on the chin...

On Wed, Sep 16, 2015 at 5:34 PM, Abhi Shah <notifications@github.com> wrote:

> In pkg/api/endpoints/util.go
> <https://github.com/kubernetes/kubernetes/pull/13778#discussion_r39688300>
> :
>
> > @@ -83,21 +65,63 @@ func RepackSubsets(subsets []api.EndpointSubset) []api.EndpointSubset {
> >  	// Next, build the N-to-M association the API wants.
> >  	final := []api.EndpointSubset{}
> >  	for key, ports := range addrSetsToPorts {
> > -		addrs := []api.EndpointAddress{}
> > -		for k := range addrSets[key] {
> > -			addrs = append(addrs, *k)
> > +		var addrs, notReadyAddrs []api.EndpointAddress
> > +		for k, ready := range addrSets[key] {
>
> can you rename k to address
> all the single-letter variables in this function and the joinAddress
> function need to be renamed.
> it is very difficult to follow what is going on...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13778/files#r39688300>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift